src/components/Company/PaySchedule/_parts/Edit.tsx

```diff
@@ -92,26 +107,50 @@ export const Edit = () => {
           </Flex>
         </div>
         <Flex flexDirection="column" gap={4} justifyContent="center" alignItems="center">
-          {!payPreviewLoading &&
-            payPeriodPreview &&
-            payPeriodPreview.map((payPeriod, index) => {
-              if (index >= 3) {
-                return
+          {payPeriodPreview && payPeriodPreview[selectedPayPeriodIndex] && (
+            <CalendarDisplay
+              key={selectedPayPeriodIndex}
+              selectionControl={
+                !payPreviewLoading && (
+                  <Select
+                    control={control}
+                    name="pay_period_preview_range"
+                    label={t('labels.preview')}
+                    items={payPeriodPreview.map((period, index) => {
+                      return {
+                        id: index,
+                        name: `${period.start_date ? new Date(period.start_date).toLocaleDateString('en-US', { weekday: 'short', month: 'short', day: 'numeric' }) : ''} – ${period.end_date ? new Date(period.end_date).toLocaleDateString('en-US', { weekday: 'short', month: 'short', day: 'numeric' }) : ''}`,
+                      }
+                    })}
+                    defaultSelectedKey={selectedPayPeriodIndex}
+                    onSelectionChange={value => {
+                      if (typeof value === 'number') {
+                        setSelectedPayPeriodIndex(value)
+                      }
+                    }}
+                  >
+                    {option => <ListBoxItem key={option.id}>{option.name}</ListBoxItem>}
+                  </Select>
+                )
               }
-              return (
-                <PayPreviewCard
-                  key={index}
-                  checkdate={new Date(payPeriod.check_date ?? '')}
-                  endDate={new Date(payPeriod.end_date ?? '')}
-                  startDate={new Date(payPeriod.start_date ?? '')}
-                  runPayrollBy={new Date(payPeriod.run_payroll_by ?? '')}
-                />
-              )
-            })}
-          {payPreviewLoading && (
-            <Flex justifyContent={'center'} alignItems={'center'}>
-              <Spinner title={t('loading')} />
-            </Flex>
+              rangeSelected={{
+                start: payPeriodPreview[selectedPayPeriodIndex].start_date || '',
+                end: payPeriodPreview[selectedPayPeriodIndex].end_date || '',
+                label: t('payPreview.payPeriod') || 'Pay Period',
+              }}
+              highlightDates={[
+                {
+                  date: payPeriodPreview[selectedPayPeriodIndex].check_date || '',
+                  highlightColor: 'black',
```

Comment:
Nit: since the consumer isn't going to have access to these props when they are being used deep inside these PaySchedule components i think the highlightColor should be set at the theming level so that it can be customized.

Reply:
Let me know if the theming i've done here covers it